crates/plugins/mediator-coordination/src/forward/routing.rs

```diff
@@ -0,0 +1,183 @@
+use std::clone;
+
+use axum::response::{IntoResponse, Response};
+use didcomm::Message;
+
+use hyper::StatusCode;
+use mongodb::bson::doc;
+use serde_json::{from_value, json, Value};
+
+use crate::{
+    model::stateful::{coord::KeylistEntry, entity::Messages},
+    web::{error::MediationError, AppState, AppStateRepository},
+};
+
+/// mediator receives messages of type forward then it unpacks the messages and stores it for pickup
+/// the unpacked message is then repacked for further transmission.
+pub async fn mediator_forward_process(
+    state: &AppState,
+    payload: Message,
+) -> Result<Option<Message>, Response> {
+    let AppStateRepository {
+        message_repository,
+        connection_repository,
+        ..
+    } = state
+        .repository
+        .as_ref()
+        .ok_or_else(|| MediationError::RepostitoryError)
+        .unwrap();
+
+    let body: Value = json!(payload.body.as_object());
+    let next: Vec<String> = from_value(body.get("next").unwrap().to_owned()).unwrap();
```

Comment:
Beware of unwrapping things carelessly. Is the next field necessarily an array?

Reply:
@Hermann-Core and i discussed it and its actually not a vector but a single value